src/ERC20/implementation/ERC20WrapperLocked.sol

```diff
@@ -97,35 +117,23 @@ abstract contract ERC20WrapperLocked is EVCUtil, Ownable, ERC20Wrapper {
         }
     }
 
-    /// @notice Sets the whitelist status for an account
-    /// @dev If the account is being whitelisted, all the locked amounts are removed resulting in all the tokens being
-    /// unlocked. If the account being removed from the whitelist, the current account balance is locked. The side
-    /// effect of this behavior is that the owner can modify the lock schedule for users, i.e. by adding and then
-    /// removing an account from the whitelist, the owner can reset the unlock schedule for that account. It must be
-    /// noted though that the ability to modify whitelist status and its effects on locks is a core feature of this
-    /// contract.
+    /// @notice Sets the whitelist status for a specified account
     /// @param account The address to set the whitelist status for
-    /// @param status The whitelist status to set
-    function setWhitelistStatus(address account, bool status) public onlyEVCAccountOwner onlyOwner {
-        if (isWhitelisted[account] != status) {
-            if (status) {
-                EnumerableMap.UintToUintMap storage map = lockedAmounts[account];
-                uint256[] memory lockTimestamps = map.keys();
-                for (uint256 i = 0; i < lockTimestamps.length; ++i) {
-                    map.remove(lockTimestamps[i]);
-                    emit LockRemoved(account, lockTimestamps[i]);
-                }
+    /// @param status The new whitelist status to set
+    function setWhitelistStatus(address account, WhitelistStatus status) public onlyEVCAccountOwner onlyOwner {
+        if (whitelistStatus[account] != status) _setWhitelistStatus(account, status);
+    }
+
+    /// @notice Allows a whitelisted account to degrade its own whitelist status
+    /// @param status The new whitelist status to set
+    function setWhitelistStatus(WhitelistStatus status) public onlyWhitelisted(_msgSender()) {
```

Comment:
Is there a use-case for needing to degrade your status?

Reply:
yes. if you have the lower whitelist status and have some tokens, you're unable to do anything with them yourself. you cannot unwrap, you can only transfer them to a non-whitelisted address to activate the lock or to the address with a higher whitelist status. only if you degrade your status, the tokens can stay with you. I don't think it will be used feature but it makes the code complete